tests/framework/db/ColumnSchemaTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace yiiunit\framework\db;
+
+use yiiunit\framework\db\DatabaseTestCase;
+use yii\db\ColumnSchema;
+
+/**
+ * @group db
+ * @group mysql
+ */
+class ColumnSchemaTest extends DatabaseTestCase
+{
+
+    public function testDbTypecastNoCommas()
+    {
+        $locale = setlocale(LC_NUMERIC, 0);
+        if (false === $locale) {
+            $this->markTestSkipped('Your platform does not support locales.');
+        }
+
+        try {
+            // This one sets decimal mark to comma sign
+            setlocale(LC_NUMERIC, 'ru_RU.utf8');
+            $cschemaDouble = new ColumnSchema([
+                'type' => 'double',
+                'dbType' => 'double',
+                'phpType' => 'double',
+            ]);
+            $this->assertEquals('1.23', (string) $cschemaDouble->dbTypecast('1.23'));
+            $this->assertEquals('1.23', (string) $cschemaDouble->dbTypecast(1.23));
+            $this->assertEquals('2.0E+30', (string) $cschemaDouble->dbTypecast(2e+30));
```

Comment:
I am confused by the test here, you are casting to string after calling `dbTypecast()` so you are not testing the result of `dbTypecast()` but the result of the cast after it has been called. This does not make sense. If I test the result of `dbTypecast()` directly the test is working fine:

```php
            $this->assertSame(1.23, $cschemaDouble->dbTypecast('1.23'));
            $this->assertSame(1.23, $cschemaDouble->dbTypecast(1.23));
            $this->assertSame(2.0E+30, $cschemaDouble->dbTypecast(2e+30));
```

I do not see how this test is reproducing a problem with the `dbTypecast()` method.